### PR TITLE
Change train schedule endpoint batch post

### PR DESCRIPTION
**Pull request by flomonster**
Doc update: https://github.com/OpenRailAssociation/osrd-website/pull/193

### CODE REVIEW

File: editoast/src/views/v2/timetable.rs [hunk 1]
```diff
@@ -240,6 +247,46 @@ async fn delete(
...
+
+    let changesets: Vec<TrainScheduleChangeset> = data
+        .into_inner()
+        .into_iter()
+        .map(|ts| TrainScheduleForm {
+            timetable_id: Some(timetable_id),
+            train_schedule: ts,
+        })
+        .map_into()
+        .collect();
```
Comment (leovalais) [inline editoast/src/views/v2/timetable.rs:296]: It's fine like this, just to note that create_batch* functions accept an `IntoIterator<Item = Changeset>`, so you can omit collect if you want.

**Review by leovalais — COMMENTED** [review]
LGTM (1st pass)

**Review by younesschrifi — APPROVED** [review]
LGTM :)

File: front/src/common/api/osrdEditoastApi.ts [hunk 2]
```diff
@@ -3324,6 +3327,10 @@ export type SimulationSummaryResult =
         error_type: string;
       };
     };
+export type TrainScheduleForm = TrainScheduleBase & {
+  /** Timetable attached to the train schedule */
+  timetable_id?: number | null;
```
Comment (SharglutDev) [inline front/src/common/api/osrdEditoastApi.ts:3342]: timetable_id is optional/null to update a trainschedule but compulsory to post one. Is that normal ?

**Review by leovalais — APPROVED** [review]

File: front/src/common/api/osrdEditoastApi.ts (hunk 2 above)
Comment (flomonster) [reply front/src/common/api/osrdEditoastApi.ts:3342]: Yes, a train schedule must be attached to a timetable. So you have to provide one at creation (it's provided in the url as path parameter).
To patch a train schedule if the user doesn't provide a timetable_id, then the train schedule stay attach to the same timetable.
This field has been added to allow the front end to move a schedule train (without having to delete and recreate it).

Comment (SharglutDev) [reply front/src/common/api/osrdEditoastApi.ts:3342]: ok got it !

**Pull request merged**